src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -3379,12 +3374,40 @@ public override void VisitNamedType(NamedTypeSymbol symbol)
             }
         }
 
+        /// <returns><see langword="true"/> if file types are present in files with duplicate file paths. Otherwise, <see langword="false" />.</returns>
         private bool CheckDuplicateFilePaths(DiagnosticBag diagnostics)
         {
             var visitor = new DuplicateFilePathsVisitor(diagnostics);
             return visitor.CheckDuplicateFilePathsAndFree(SyntaxTrees, GlobalNamespace);
         }
 
+        /// <returns><see langword="true"/> if duplicate interceptors are present in the compilation. Otherwise, <see langword="false" />.</returns>
+        private bool CheckDuplicateInterceptions(DiagnosticBag diagnostics)
+        {
+            if (_interceptions is null)
+            {
+                return false;
+            }
+
+            var anyDuplicates = false;
```

Comment:
>var [](http://example.com/codeflow?start=12&length=3)

nit: we're not saving much characters by using `var`. Consider explicit type for primitives